File: Server/Dtos/UserDto.cs
namespace Server.Dtos
{
    public class UserDto
    {
        public Guid UserId { get; set; }
        public required string DisplayName { get; set; }
        public required string FirstName { get; set; }
        public required string LastName { get; set; }
        public required string ProfileImageUrl { get; set; }
        public required string Username { get; set; }
        public required string UserEmail { get; set; }
        public required DateTime CreatedAt { get; set; }
    }
}

File: Client/src/app/shared/models/user.dto.ts
interface UserDto {
    userId: string;
    displayName: string;
    profileImageUrl: string;
    firstName: string,
    lastName: string,
    createdAt: Date,
    username: string,
    postCount: number,
    followerCount: number,
    followingCount: number,
    IsFollowedByCurrentUser: boolean
}
